metrics/run_all_metrics.sh

```diff
@@ -31,32 +31,104 @@ echo "Check dependencies"
 cmds=("docker" "bc" "awk" "smem" "ab")
 check_cmds "${cmds[@]}"
 
-echo "Running all metrics tests"
-
-# Run the time tests
-bash ${SCRIPT_PATH}/time/docker_workload_time.sh true busybox $RUNTIME 100
-
-# Run the network metrics
-bash ${SCRIPT_PATH}/network/network-latency.sh
-bash ${SCRIPT_PATH}/network/network-metrics-cpu-consumption.sh
-bash ${SCRIPT_PATH}/network/network-metrics-iperf3.sh
-bash ${SCRIPT_PATH}/network/network-metrics-memory-pss-1g.sh
-bash ${SCRIPT_PATH}/network/network-metrics-memory-pss.sh
-bash ${SCRIPT_PATH}/network/network-metrics-memory-rss-1g.sh
-bash ${SCRIPT_PATH}/network/network-metrics-nuttcp.sh
-bash ${SCRIPT_PATH}/network/network-metrics.sh
-bash ${SCRIPT_PATH}/network/network-nginx-ab-benchmark.sh
-
-# Run the density/footprint scripts
-# If you have KSM enabled, the 'settle time' depends on the default settings.
-# For instance, for an Ubuntu 16.04 default install, it takes about 200s for
-# 20 CC containers footprint to 'settle down'.
-#
-# Presume KSM is off, and hence no need to have a settle time
-bash ${SCRIPT_PATH}/density/docker_memory_usage.sh 20 1
-
-# Run I/O storage tests
-bash ${SCRIPT_PATH}/storage/fio_job.sh -b 16k -o randread -t "storage IO random read bs 16k"
-bash ${SCRIPT_PATH}/storage/fio_job.sh -b 16k -o randwrite -t "storage IO random write bs 16k"
-bash ${SCRIPT_PATH}/storage/fio_job.sh -b 16k -o read -t "storage IO linear read bs 16k"
-bash ${SCRIPT_PATH}/storage/fio_job.sh -b 16k -o write -t "storage IO linear write bs 16k"
+function help() {
+echo "$(cat << EOF
+Usage: $0 [-h] [--help] [-v] [--version]
+   Description:
+        This script allows to execute the metrics tests
+        in different groups which are focused in different
+        aspects of the containers. If no option is specified
+        all tests will be executed.
+   Options:
+        -h, --help         Shows help
+        -l, --latency      Run latency/time metrics tests
+        -m, --memory       Run memory metrics tests
+        -n, --network      Run network metrics tests
+        -s, --storage      Run I/O storage metrics tests
+EOF
+)"
+}
+
+# Only run latency metrics tests
+function run_latency_tests() {
+	# Run the time tests
+	bash ${SCRIPT_PATH}/time/docker_workload_time.sh true busybox $RUNTIME 100
+}
+
+# Only run network metrics tests
+function run_network_tests() {
```

Comment:
maybe good to split the network tests into network minimum and full. They take a long time to run. Most of the time we care only about iperf3

Reply:
then iperf3 tests could be "minimum" tests group?

Reply:
good point @mcastelino  In fact, we could split all the groups (density, time, io, network) into two groups each - 'core' and 'full'. We may then be able to replace https://github.com/clearcontainers/tests/blob/master/.ci/run_metrics_ci.sh with a call to this script for 'core', and remove some duplication.

Reply:
For reference, our current 'core' network tests look like (from the ci script)...
```
# ops/second
bash network/network-nginx-ab-benchmark.sh
# ping latency
bash network/network-latency.sh
# Bandwidth and jitter
bash network/network-metrics-iperf3.sh
# UDP bandwidths and packet loss
bash network/network-metrics-nuttcp.sh
```

@mcastelino - if you think we should add/remove from that core network group then feel free to comment.

Reply:
Note: if we do move to splitting the functions and then invoking this script for the metrics CI, we will have to work out how to take care of KSM on/off testing of memory footprint - the current ci script does do that, and invokes the memory test with different parameters ('settle time') depending on the KSM mode for instance.